pkg/apis/extensions/types.go

```diff
@@ -405,7 +405,7 @@ type JobSpec struct {
 	Completions *int `json:"completions,omitempty"`
 
 	// Selector is a label query over pods that should match the pod count.
-	Selector map[string]string `json:"selector"`
+	Selector PodSelector `json:"selector"`
```

Comment:
Should be optional: make it a pointer and add omitempty.

We'll set it from the pod template by default.